crates/html-macro/src/parser/statement.rs

```diff
@@ -0,0 +1,80 @@
+use crate::parser::HtmlParser;
+use quote::quote;
+use syn::{Stmt, Expr, ExprIf};
+
+impl HtmlParser {
+    /// Parse an incoming syn::Stmt node inside a block
+    pub(crate) fn parse_statement(
+        &mut self,
+        stmt: &Stmt,
+    ) {
+        // Here we handle a block being a descendant within some html! call.
+        //
+        // The descendant should implement Into<IterableNodes>
+        //
+        // html { <div> { some_node } </div> }
+        match stmt {
+            Stmt::Expr(expr) => self.parse_expr(stmt, expr),
+            _ => self.push_iterable_nodes(stmt, None)
+        };
+    }
+
+    /// Parse an incoming syn::Expr node inside a block
+    pub(crate) fn parse_expr(
+        &mut self,
+        stmt: &Stmt,
+        expr: &Expr
+    ) {
+        match expr {
+            Expr::If(expr_if) => {
+                self.expand_if(stmt, expr_if);
+            },
+            _ => {
+                self.push_iterable_nodes(stmt, None);
+            }
+        }
+    }
+
+    /// Expand an incoming Expr::If block
+    /// This enables us to use JSX-style conditions inside of blocks such as
+    /// the following example.
+    /// 
+    /// # Examples
+    /// 
+    /// ```rust,ignore
+    /// html! {
+    ///     <div>
+    ///         {if condition_is_true {
+    ///	            html! { <span>Hello World</span> }
+    ///         }}
+    ///     </div>
+    /// }
+    /// ```
+    /// 
+    /// Traditionally this would be possible as an if statement in rust is an
+    /// expression, so the then, and the else block have to return matching types.
+    /// Here we identify whether the block is missing the else and fill it in with
+    /// a blank VirtualNode
```

Comment:
I know that method is small so it's easy to see - but can we still specify `a blank VirtualNode::text("")` just so it's slightly more clear what blank means